### PR TITLE
feat: migrate f36-forms/controlled-input-field to v4 [HOMER-25] 

### PR DESCRIPTION
**Pull request by Chaoste**
# Purpose of PR

Extract the ControlledInputField component from react-components and move it to a new package f36-forms. Afterwards we can move RadioButtonField and CheckboxField.

- Create f36-forms package
- Move ControlledInputField from f36-react-components to f36-forms
- Migrate CSS to emotion

- [x] Evaluate !important css rules

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information
- [x] FormLabel is migrated to emotion

### CODE REVIEW

**Comment by denkristoffer** [comment]
@domarku @fabe Can you review the design on Chromatic please?

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (denkristoffer) [inline packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:12]: You should extend BoxProps here instead to get the margin and padding props.

File: packages/forma-36-react-components/.size-limit
Comment (denkristoffer) [inline packages/forma-36-react-components/.size-limit:6]: What's causing such a size increase?

**Review by denkristoffer — CHANGES_REQUESTED** [review]

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (Chaoste) [reply packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:12]: Thanks, will do so :)

File: packages/forma-36-react-components/.size-limit
Comment (Chaoste) [reply packages/forma-36-react-components/.size-limit:6]: I assume this is somehow related to creating a new package?

**Comment by Chaoste** [comment]
Hi @fabe , I was also wondering why the font size changed. It appears that this is caused by FormLabel not being written in emotion so far. We have another ticket to do this.
Since we will publish the whole v4 at once, I’d say it’s okay to merge this one now and migrate the FormLabel afterwards to not block the process too much. There are three pending PRs and I don’t know if I can spare time next week to support in f36 v4 migration by taking over another ticket.

File: packages/components/forms/src/controlled-input-field/ControlledInputField.styles.ts
Comment (Chaoste) [inline packages/components/forms/src/controlled-input-field/ControlledInputField.styles.ts:5]: This also needs to be applied to the label within ControlledInput to have the exact same styles as before (which can be seen in chromatic). PR to enable passing styles down to the label in ControlledInput is upcoming.

Comment (Chaoste) [reply packages/components/forms/src/controlled-input-field/ControlledInputField.styles.ts:5]: @denkristoffer @fabe feel free to review [this mini PR](https://github.com/contentful/forma-36/pull/977) (which blocks this one)

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (denkristoffer) [inline packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:41]:
```suggestion
```

Is this used?

**Review by denkristoffer — CHANGES_REQUESTED** [review]

File: packages/components/forms/stories/CheckboxField.stories.tsx
Comment (denkristoffer) [inline packages/components/forms/stories/CheckboxField.stories.tsx:4]: Is this not exported?

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (denkristoffer) [inline packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:109]: I guess we should rename this prop...

**Review by denkristoffer — CHANGES_REQUESTED** [review]

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (Chaoste) [reply packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:41]: Good point, the internal props don't contain a `ref` attribute. They can't since it will be captured by `forwardRef` and excluded from the `props` object. So for external usage, it's correct that we handle a `ref` prop while it's internally not really part of `props`. Let's just remove the `ref` definition from this type. I saw we currently have multiple approaches to export a PropsType, so we probably refactor all of them at some point at once.

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (Chaoste) [reply packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:108]: Yes, the guidelines point out that one should always prefix booleans with `is` (or something similar). But it's not part of this Part since it's only related to the `Label` :)

Comment (Chaoste) [reply packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:108]: I also want to rename the `/Label` to `/label` since all folders should be lowerCase but I will tackle those small refinements in a future small PR (no big PRs anymore xD)

File: packages/components/forms/stories/CheckboxField.stories.tsx
Comment (Chaoste) [reply packages/components/forms/stories/CheckboxField.stories.tsx:4]: It is, autofix didn't work out as expected.

File: packages/forma-36-react-components/src/components/RadioButtonField/RadioButtonField.test.tsx
Comment (Chaoste) [reply packages/forma-36-react-components/src/components/RadioButtonField/RadioButtonField.test.tsx:11]: I don't know why. The radio button field users the controlled input field which uses the controlled *input* (first inputProps) which again renders a native *input* DOM element (second inputProps). The prop objects are not the same. I don't like it either and was hoping that this test code might reveal this curiosity for other developers and improve it at a later point.
If you have a proposal, feel free to speak up :P

File: packages/forma-36-react-components/src/components/RadioButtonField/RadioButtonField.tsx
Comment (Chaoste) [reply packages/forma-36-react-components/src/components/RadioButtonField/RadioButtonField.tsx:5]: There is a separate PR for the radio button field which also [removes this line](https://github.com/contentful/forma-36/pull/961/files#diff-c0ebc14c0a50ec758adf0a1d2e8d715d6724ff5baf7153652886a8b342a6ae96R5). Let's merge this one first and directly afterwards merge the other one.

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (Chaoste) [reply packages/components/forms/src/controlled-input-field/ControlledInputField.tsx]: Ah, now I remember that removing this causes a typescript error. There are different definitions of ReactRef, a legacy one, and a more recent one and they are not compatible...
```
    // Used by forwardRef
    interface RefAttributes<T> extends Attributes {
        ref?: Ref<T>;
    }
    // Used by BoxProps
    interface ClassAttributes<T> extends Attributes {
        ref?: LegacyRef<T>;
    }
```

File: packages/components/forms/src/controlled-input-field/ControlledInputField.tsx
Comment (denkristoffer) [reply packages/components/forms/src/controlled-input-field/ControlledInputField.tsx:109]: Sure as long as we keep it consistent, there are a few other places to rename then.

File: packages/forma-36-react-components/src/components/RadioButtonField/RadioButtonField.test.tsx
Comment (denkristoffer) [reply packages/forma-36-react-components/src/components/RadioButtonField/RadioButtonField.test.tsx:11]: I don't think we should expose native input props, `style` should work with the controlled input field itself.

**Comment by denkristoffer** [comment]
There's a margin difference in the Basic story that I can't see why is there. Thoughts?

**Review by denkristoffer — APPROVED** [review]
🙏

**Pull request merged**
